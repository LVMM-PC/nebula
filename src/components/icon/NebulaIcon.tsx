import { Component, Prop, Vue } from "vue-property-decorator";

@Component({
  name: "NebulaIcon"
})
export default class NebulaIcon extends Vue {
  @Prop({ default: null, type: String })
  private type?: string;

  @Prop({ default: null, type: String })
  private title?: string;

  @Prop({ default: false, type: Boolean })
  private spin?: boolean;

  @Prop()
  private timeout?: any;

  public prefixCls: string = "nebula-icon";

  public clicked: boolean = false;

  get classes() {
    let prefixCls = this.prefixCls;
    let type = this.type;
    let spin = this.spin || (type && type.indexOf("loading") === 0);
    return [
      prefixCls,
      {
        [`${prefixCls}-${type}`]: type,
        [`${prefixCls}-spin`]: spin
      }
    ];
  }

  handleClick(event) {
    if (this.clicked) {
      return;
    }

    this.clicked = true;
    clearTimeout(this.timeout);
    this.timeout = setTimeout(() => (this.clicked = false), 500);
    this.$emit("click", event);
  }

  render() {
    const { title, classes, handleClick, $listeners } = this;
    const iconProps = {
      attrs: {
        title
      },
      class: classes,
      on: {
        ...$listeners,
        click: handleClick
      }
    };
    return <i {...iconProps} />;
  }
}
