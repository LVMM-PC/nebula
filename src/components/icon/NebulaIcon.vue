<template>
  <span
    v-bind="$attrs"
    :class="classes"/>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class NebulaIcon extends Vue {
  @Prop({ default: null, type: String })
  private type?: string;

  @Prop({ default: false, type: Boolean })
  private spin?: boolean;

  public prefixCls: string = "nebula-icon";

  get classes() {
    let prefixCls = this.prefixCls;
    let type = this.type;
    let spin = this.spin || (type && type.indexOf("loading") === 0);
    return [
      prefixCls,
      {
        [`${prefixCls}-${type}`]: type,
        [`${prefixCls}-spin`]: spin
      }
    ];
  }
}
</script>
