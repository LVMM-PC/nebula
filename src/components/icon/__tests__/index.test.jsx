import { mount } from "@vue/test-utils";
import NebulaIcon from "../NebulaIcon";

describe("Icon", () => {
  it("should render to a <span class='???'></span>", () => {
    const wrapper = mount({
      render() {
        return <NebulaIcon type="appstore" class="my-icon-class-name" />;
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
