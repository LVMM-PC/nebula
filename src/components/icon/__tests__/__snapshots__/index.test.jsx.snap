// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render to a <span class='???'></span> 1`] = `<span class="nebula-icon nebula-icon-appstore my-icon-class-name"></span>`;
